auth-web/src/components/auth/create-account/PaymentMethodSelector.vue

```diff
@@ -101,14 +121,40 @@ export default class PaymentMethodSelector extends Mixins(Steppable) {
     this.isPADValid = isValid
   }
 
-  private save () {
+  private async save () {
+    this.isLoading = true
     this.setCurrentOrganizationPaymentType(this.selectedPaymentMethod)
-    this.createAccount()
+    try {
+      const bcolAccountDetails = await this.validateBcolAccount(this.currentOrganization.bcolProfile)
+      this.errorMessage = bcolAccountDetails ? null : 'Error - No account details provided for this account.'
+      this.setCurrentOrganizationBcolProfile(this.currentOrganization.bcolProfile)
+    } catch (err) {
+      switch (err.response.status) {
+        case 409:
+          this.errorMessage = err.response.data.message
+          break
+        case 400:
+          this.errorMessage = err.response.data.message
+          break
+        default:
+          this.errorMessage = 'An error occurred while attempting to create your account.'
+      }
+    } finally {
+      this.isLoading = false
+    }
+    if (!this.errorMessage) {
+      this.createAccount()
+    }
   }
 
   @Emit('final-step-action')
   private createAccount () {
   }
+
+  @Emit('emit-bcol-info')
+  private getBcolInfo (bcolProfile: BcolProfile) {
```

Comment:
does it make more sense to call this `setBcolInfo` ?

Reply:
I think so, ideally I'd like to get rid of these emits